hack/build_speed.sh

```diff
@@ -0,0 +1,134 @@
+#! /bin/sh
+
+# The main goal of this script is test and time builds using Buildah or Docker.
+# We hope to use it to help optimize Buildah build performance
+#
+# It takes two options
+# First option tells the type of the container image
+# build to do.  Valid options are:
+#       Docker  - docker build
+#       Buildah - buildah bud
+#       Both    - Do docker build followed by buildah bud
+#
+# Second Option specifies a directory or cleanup
+# The script will 'find' files beginning with Dockerfile, for each Dockerfile
+# it finds it will run a build with the Dockerfile and directory for the
+# context. When it does the builds, it will call time on them to show how
+# long the builds take.  The created image name will be a combination of the
+# lowercased Directory name that the Dockerfile was found in plus the lower
+# cased dockerfile name.
+#
+# if the second field is cleanup, the script will remove all images from the
+# specified builder.
+#
+# The script does not check for conflicts on nameing.
+#
+# Outputs file:
+#
+#
+# cat /tmp/build_speed.json
+# {
+#   "/usr/share/fedora-dockerfiles/redis/Dockerfile": {
+#     "docker": {
+#       "command": "docker build -f /usr/share/fedora-dockerfiles/redis/Dockerfile -t redis_dockerfile /usr/share/fedora-dockerfiles/redis",
+#       "real": "3:28.70"
+#     },
+#     "buildah": {
+#       "command": "buildah bud --layers -f /usr/share/fedora-dockerfiles/redis/Dockerfile -t redis_dockerfile /usr/share/fedora-dockerfiles/redis",
+#       "real": "2:55.48"
+#     }
+#   }
+# }
+#
+# Examples uses
+#     ./build_speed.sh Docker ~/MyImages
+#     ./build_speed.sh Both /usr/share/fedora-dockerfiles/django/Dockerfile
+
+#totalsfile=$(mktemp /tmp/buildspeedXXX.json)
+totalsfile=/tmp/build_speed.json
+echo -n '{'>$totalsfile
+commaDockerfile=""
+
+Dockerfiles() {
+    find -L $1 -name Dockerfile\*
+}
+
+Buildah() {
+    Name=$1
+    Dockerfile=$2
+    Context=$3
+    echo buildah bud --layers -f ${Dockerfile} -t ${Name} ${Context}
+    Time buildah bud --layers -f ${Dockerfile} -t ${Name} ${Context}
+}
+
+Time() {
+    outfile=$(mktemp /tmp/buildspeedXXX)
+    /usr/bin/time -o $outfile --f "%E" $@
+    echo "{\"engine\": \"$1\", \"command\": \"$@\", \"real\": \"$(cat ${outfile})\"}"
+    echo ${comma}  >> $totalsfile
+    if [ ${comma} == "" ]; then
+	comma=","
+    fi
+    echo -n "\"$1\": {\"command\": \"$@\", \"real\": \"$(cat ${outfile})\"}" >> $totalsfile
```

Comment:
What I was trying for is simplicity: there's no need to `if` anything, and no need for extra echo lines. You can replace lines 68-72 with:
```
    echo -n "${comma}\"$1\":....
    comma=","
```
i.e. the comma is printed on all iterations except the first one. (you may want to initialize `comma=` around line 50 as good practice).

Reply:
done